src/umpire/Umpire.hpp

```diff
@@ -122,6 +122,21 @@ bool pointer_overlaps(void* left, void* right);
  */
 bool pointer_contains(void* left, void* right);
 
+/*!
+ * \brief Check whether or not an Allocator is accessible from a given platform.
+ *
+ *\param camp::Platform p  
+ *\param umpire::Allocator a 
+ */
+bool isAccessible(Platform p, Allocator a);
+
+/*!
+ * \brief Helper function to find the resource type of a given allocator
+ *
+ *\param umpire::allocator a  
+ */
+MemoryResourceTraits::resource_type findResource(Allocator a);
```

Comment:
If you want to keep this public, can you rename to `get_resource`. I think that "find" makes it sound like it has to search for something, but we already know the resource.

Alternatively, you could move this into an anonymous namespace in Umpire.cpp so it won't be exposed to users.